ex02.exs

```diff
@@ -81,23 +78,26 @@ end
 defmodule Ex02 do
 
   def new_counter(initial_value \\ 0) do
-    # « your code goes here»
+    {:ok, myagent} = Agent.start_link(fn -> initial_value end)
+    myagent
   end
 
   def next_value(agent) do
-    # « your code goes here»
+    nextvalue = Agent.get_and_update(agent, fn total -> {total, total + 1} end)
+    nextvalue
   end
 
   @global_name :my_global_agent
 
   def new_global_counter(initial_value \\ 0) do
-    # « your code goes here»
     # You can give an agent a name by adding the
     # `name: @global_name` option to start_link
-  end
+    {:ok, myagent} = Agent.start_link(fn -> initial_value end, [name: @global_name])
```

Comment:
This generates a warning that `myagent` is unused. I treat all warnings as errors (because the are, more often than not) so I recommend getting into the habit of getting clean compilations, both on my assignments and in general.

# -1